packages/preferences/src/browser/preference-menu-factory.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from "inversify";
+import { BrowserMainMenuFactory, DynamicMenuWidget } from "@theia/core/lib/browser/menu/browser-menu-plugin";
+import { Menu as MenuWidget} from "@phosphor/widgets";
+import { ActionMenuNode, Command, CompositeMenuNode } from "@theia/core";
+import { CommandRegistry } from "@theia/core";
+import { PreferenceProperty } from "@theia/core/lib/browser";
+
+@injectable()
+export class PreferencesMenuFactory extends BrowserMainMenuFactory {
+
+    private commandsStorage: Command[] = [];
```

Comment:
you should use `Disposable` pattern instead:
```ts
protected readonly toUnregister = new DisposableCollection();

this.toUregister.push(this.commands.registerCommand(...));

menu.aboutToClose.connect(() => this.toUnregister.dispose());
```

Reply:
done